src/zimscraperlib/filesystem.py

```diff
@@ -44,5 +46,7 @@ def get_content_mimetype(content: bytes | str) -> str:
 
 def delete_callback(fpath: str | pathlib.Path):
     """helper deleting passed filepath"""
-
-    os.unlink(fpath)
+    if not pathlib.Path(fpath).exists():
```

Comment:
This can be raced. Unless there's a valid reason to accept `str` here, it should be `Path` and we should thus use `fpath.unlink(missing_ok=True)` which is exactly what we want

Reply:
Should we still log something if missing? (this was a suggestion from the PR, and I think we not really mind about logging, especially since it can be raced anyways)

Reply:
I don't think that's necessary